### PR TITLE
feat: `Syrup` ark

### PR DESCRIPTION
**Pull request by halaprix**
I'll help you write a PR message based on the diff provided. Let me analyze the changes and create a comprehensive PR description.

## Description
This PR introduces a new SyrupArk contract that enables integration with Maple Finance's Syrup pools, allowing for token supply management and yield generation through signature-authorized deposits and managed withdrawals.

## Changes
- Added new `SyrupArk.sol` contract implementing the Ark interface for Syrup pool integration
- Created new interfaces for Syrup protocol integration:
  - `ISyrupPool.sol`
  - `ISyrupManager.sol`
  - `ISyrupWithdrawalManager.sol`
  - `ISyrupRouter.sol`
- Added comprehensive fork tests in `SyrupArk.fork.t.sol`
- Added deployment configuration for Syrup pools in `index.json`
- Created deployment module `syrup-ark.ts` for SyrupArk contract deployment

## Benefits
1. Enables integration with Maple Finance's Syrup pools, expanding the protocol's yield generation capabilities
2. Implements secure signature-based authorization for deposits
3. Provides managed withdrawal functionality through Syrup's withdrawal manager
4. Includes comprehensive testing with fork tests to ensure real-world compatibility

## Testing
The changes have been tested through:
- Fork tests that verify:
  - Successful boarding of assets with signature authorization
  - Partial and full redemption requests
  - Asset tracking in withdrawal queue
  - Interest accrual verification
- Tests are run against mainnet fork at block 22274128
- Tests cover both partial and full redemption scenarios
- Tests verify proper event emission and state changes

## Next steps
- Monitor initial deployments for any edge cases
- Consider adding additional test coverage for edge cases
- Document integration process for protocol users
- Consider adding monitoring for withdrawal queue status

## Additional Notes
- The implementation includes proper error handling and validation
- The contract maintains compatibility with existing Ark interface
- Signature authorization is required for deposits
- Withdrawals are managed through Syrup's withdrawal queue system
- Initial configuration includes USDC pool integration

Please review and provide any feedback or suggestions for improvement.

## Summary by CodeRabbit

- **New Features**
	- Introduced a referral code mechanism for deposits in Syrup Ark, using a fixed "summer" code.
	- Added new event logging for deposits, including referral code information.

- **Bug Fixes**
	- Simplified and streamlined the deposit process, removing unnecessary authorization steps.

- **Tests**
	- Updated tests to reflect the new deposit flow and referral code usage.
	- Improved test setup with more explicit permissioning and address labeling.
	- Added new tests for withdrawal asset accounting and redemption processing.

- **Documentation**
	- Updated README coverage badge to reflect a new branch.

- **Style**
	- Reformatted JSON proposal files for improved readability (no change to content).

### CODE REVIEW

File: packages/core-contracts/src/contracts/arks/SyrupArk.sol [hunk 1]
```diff
@@ -93,7 +95,7 @@ contract SyrupArk is Ark {
      * @notice Request redemption of shares from the Syrup pool
      * @param amount Amount of shares to redeem
```
Comment (zerotucks) [inline packages/core-contracts/src/contracts/arks/SyrupArk.sol:96]: Might want to update the comments here

**Review by zerotucks — APPROVED** [review]
